src/lib/DataTable/DataTable.svelte

```
@@ -0,0 +1,1357 @@
+  let pageSize = $state(pagination?.pageSize ?? 10);
+
+  const totalRows = $derived(
+    serverSidePagination ? (pagination?.totalRows ?? data.length) : processedData.length
+  );
+  const totalPages = $derived(totalPagesOf(totalRows, pageSize));
+
+  // Clamp page if the dataset shrank below the current page.
+  $effect(() => {
+    if (page > totalPages) page = totalPages;
+    if (page < 1) page = 1;
+  });
```

ESLint

[error] $effect is not allowed. Use a different reactive pattern instead.

(no-restricted-syntax)